extension/llm/runner/test/test_runner_pybindings.py

```diff
@@ -0,0 +1,269 @@
+        image = Image()
+        image.width = 640
+        image.height = 480
+        image.channels = 3
+
+        repr_str = repr(image)
+        self.assertIn("Image", repr_str)
+        self.assertIn("height=480", repr_str)
+        self.assertIn("width=640", repr_str)
+        self.assertIn("channels=3", repr_str)
+
+
+class TestMultimodalInput(unittest.TestCase):
+    """Test the MultimodalInput class."""
+
+    def test_text_input(self):
+        """Test creating a text MultimodalInput."""
+        # Test direct constructor
+        text_input = MultimodalInput("Hello, world!")
+        self.assertTrue(text_input.is_text())
+        self.assertFalse(text_input.is_image())
+        self.assertEqual(text_input.get_text(), "Hello, world!")
+
+        # Test helper function
+        text_input2 = make_text_input("Test text")
+        self.assertTrue(text_input2.is_text())
+        self.assertEqual(text_input2.get_text(), "Test text")
+
+    def test_image_input(self):
+        """Test creating an image MultimodalInput."""
+        # Create an image
+        image = Image()
+        image.data = [255] * (100 * 100 * 3)
+        image.width = 100
+        image.height = 100
+        image.channels = 3
+
+        # Test direct constructor
+        image_input = MultimodalInput(image)
+        self.assertTrue(image_input.is_image())
+        self.assertFalse(image_input.is_text())
+
+        # Test helper function with numpy array
+        img_array = np.ones((50, 60, 3), dtype=np.uint8) * 128
+        image_input2 = make_image_input(img_array)
+        self.assertTrue(image_input2.is_image())
+        self.assertFalse(image_input2.is_text())
+
+    def test_invalid_image_array(self):
+        """Test error handling for invalid image arrays."""
+        # Wrong dimensions
+        with self.assertRaises(RuntimeError) as cm:
+            make_image_input(np.ones((100,), dtype=np.uint8))
+        self.assertIn("3-dimensional", str(cm.exception))
+
+        # Wrong number of channels
+        with self.assertRaises(RuntimeError) as cm:
+            make_image_input(np.ones((100, 100, 2), dtype=np.uint8))
+        self.assertIn("3 (RGB) or 4 (RGBA)", str(cm.exception))
+
+    def test_repr(self):
+        """Test string representation."""
+        # Text input
+        text_input = MultimodalInput("This is a test")
+        repr_str = repr(text_input)
+        self.assertIn("MultimodalInput", repr_str)
+        self.assertIn("type=text", repr_str)
+        self.assertIn("This is a test", repr_str)
+
+        # Long text input (should be truncated)
+        long_text = "a" * 100
+        text_input2 = MultimodalInput(long_text)
+        repr_str2 = repr(text_input2)
+        self.assertIn("...", repr_str2)
+
+        # Image input
+        image = Image()
+        image_input = MultimodalInput(image)
+        repr_str3 = repr(image_input)
+        self.assertIn("type=image", repr_str3)
+
+
+class TestMultimodalRunner(unittest.TestCase):
```

Comment:
Hmm I guess we can't really test this further, can't mock anything either. Can we add an E2E test using a sample model to show that it works in this PR

Reply:
Yeah make sense. Let me change the llava CI to do this.